sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -1057,5 +1059,19 @@ case class ScalaUDF(
 
   private[this] val converter = CatalystTypeConverters.createToCatalystConverter(dataType)
 
-  override def eval(input: InternalRow): Any = converter(f(input))
+  val npeErrorMessage = "Given UDF throws NPE during execution, please check the UDF " +
+    "to make sure it handles null parameters correctly."
+
+  override def eval(input: InternalRow): Any = {
+    val result = try {
+      f(input)
+    } catch {
+      case e: NullPointerException =>
```

Comment:
It is a bit hacky to set stack trace like this.
npe.setStackTrace(e.getStackTrace)

If user search the code line reported in the stack trace, user may not able to find the code that matches the error message.